test/unit/invoice_test.rb

```diff
@@ -119,10 +119,12 @@ def setup
   end
 
   test 'validates presence of valid period' do
-    @invoice.update(period: 'invalid')
+    %w[invalid 2023abc-01 2023-05xxx 2023-05123456789].each do |value|
+      @invoice.update(period: value)
 
-    assert_not @invoice.valid?
-    assert_includes @invoice.errors[:period], 'Billing period format should be YYYY-MM'
+      assert_not @invoice.valid?
+      assert_includes @invoice.errors[:period], 'Billing period format should be YYYY-MM'
+    end
```

Comment:
A small thing: for this, I think it's better to add a test for each case, that is:

```
%w[invalid 2023abc-01 2023-05xxx 2023-05123456789].each do |value|
  test "validates presence of #{value}" do
```